src/rust/engine/ui/src/display.rs

```diff
@@ -72,7 +72,8 @@ pub struct EngineDisplay {
   printable_msgs: VecDeque<PrintableMsg>,
   cursor_start: (u16, u16),
   terminal_size: (u16, u16),
-  async_stdin: AsyncReader,
+  async_stdin: Option<AsyncReader>,
+  suspended: bool,
```

Comment:
It looks a lot like `async_stdin.is_some()` and `suspended` will always return the same value. Could we just use `async_stdin` as a source of truth and remove `suspended`?

Reply:
I thought about this, but decided against it because the logical source of truth is whether or not the display is in a suspended state, which is a boolean property. Having async_stdin be Optional is a side-effect of that, necessary to get the dropping/recreating behavior we want, it's not the actual source of truth.